data/lua/explode_ability.lua

```diff
@@ -0,0 +1,38 @@
+
+local _ = wesnoth.textdomain 'wesnoth-help'
+local T = wml.tag
+
+-- The explode event code
+wesnoth.game_events.add_repeating("die", function()
+	local ec = wesnoth.current.event_context
+
+	if not ec.x1 or not ec.y1 then
+		return
+	end
+
+	local u_exploder = wesnoth.units.get(ec.x1, ec.y1)
+
+	if not u_exploder or not u_exploder:matches { ability = "explode" } then
+		return
+	end
+
+	local u_exploder_cfg = u_exploder.__cfg
+        local adjacent_units = wesnoth.units.find_on_map{T.filter_adjacent { id = u_exploder_cfg.id}}
+
+	for i,v in ipairs(wml.get_child(u_exploder_cfg, "abilities")) do
+		if v[1] == "dummy" and v[2].id == "explode" then
+			local dmg = v[2].value
```

Comment:
In the schema, you declared this value as an `f_int`. This means that the user can specify an integer, _or_ they can specify a formula enclosed in parentheses, for example `(unit.hitpoints / 2 + other_unit.hitpoints / 3)`.

If this is indeed the intended functionality, then you need to handle it as such. It would look something like this (adapted from similar code in random_placement.lua):

```suggestion
			local dmg = v.contents.value
			if type(dmg) == 'string' then
				if dmg:match('^%s*%(.*%)%s*$') then
					local params = {
						unit = u_exploder,
						other_unit = wesnoth.units.get(ec.x2, ec.y2)
					}
					local result = wesnoth.eval_formula(dmg, params)
					-- Now to convert it to an integer
					-- Pick only one of the following 2 lines
					dmg = math.floor(tonumber(result))
					--dmg = math.round(tonumber(result))
			else
				wml.error('invalid formula')
			end
		end
```

Alternatively, if you don't want to allow that, you can change the schema type to just `int`. This makes it inconsistent with built-in abilities, though.